### DISCUSSION TITLE
New Rule to prevent param mutation

### DISCUSSION THREAD

**Discussion by ghost** [eslint/eslint#14796] [Ideas]
'no-param-reassign' is very helpful, but doesn't prevent non assignment methods of mutation. Specifically it doesn't warn against Array.prototype mutating methods, such as `push`, `pop` and others. 

I thought about suggesting a new rule under issues, but according to the template you can only suggest rules for new ECMA spec from the last year.

**Comment by nzakas** [comment]
That’s correct. We also don’t create rules that require type information. However, you can certainly create your own rule that does this if you want.
